TypeCobol.Analysis/Dfa/DefaultDataFlowGraphBuilder.cs

```diff
@@ -11,6 +12,40 @@ namespace TypeCobol.Analysis.Dfa
     /// </summary>
     public class DefaultDataFlowGraphBuilder : DataFlowGraphBuilder<Node, VariableSymbol>
     {
+        /// <summary>
+        /// Collector of DataDefinition Nodes in the list of Instructions of a given block.
+        /// </summary>
+        public class DataDefinitionCollector : AbstractSymbolAndTypeVisitor<BasicBlock<Node, DfaBasicBlockInfo<VariableSymbol>> , BasicBlock<Node, DfaBasicBlockInfo<VariableSymbol>> >
```

Comment:
the `AbstractSymbolAndTypeVisitor` is not required. You can collect nodes using a recursive collector method and the `Node.Children` property.

Something like:

```csharp
public override HashSet<VariableSymbol> GetDeclarations()
{
    var dataDivision = this.Cfg.ProcedureDivisionNode.Parent.GetChildren<DataDivision>().SingleOrDefault();
    return new HashSet<VariableSymbol>(Collect(dataDivision));

    IEnumerable<VariableSymbol> Collect(Node node)
    {
        if (node.SemanticData is VariableSymbol variable && variable.Value != null)
        {
            yield return variable;
        }

        if (node.ChildrenCount > 0)
        {
            foreach (var child in node.Children)
            {
                foreach (var childVariable in Collect(child))
                {
                    yield return childVariable;
                }
            }
        }
    }
}
```

This requires some adaptation to add the node into rootBlock instruction list but you get the idea 😄.

Reply:
Ok

Reply:
Ok thanks, done in e9f226e